### DISCUSSION TITLE
How to replace a texture/material within a scenegraph

### DISCUSSION THREAD

**Discussion by matthew-rimmer** [visgl/luma.gl#2235] [Q&A]
Hello!

So I have a task that requires me to replace textures within a loaded GLTF file. I was kind of confused about how to go about this in luma.gl, and I was hoping to do this in luma as we already use deck.gl.

Say for this [sample model from khronos](https://github.com/KhronosGroup/glTF-Sample-Assets/blob/main/Models/BoxTextured/glTF-Binary/BoxTextured.glb), load it:

```
// Get the image and model
const testImage = await load("https://picsum.photos/300", ImageLoader, { image: { type: "data", decode: true } });
const rawModel = await load(model, GLTFLoader);

// Process
const processed = postProcessGLTF(rawModel);
luma.registerDevices([WebGLDevice]);
const device = await luma.createDevice({ type: "webgl" });
const { scenes } = createScenegraphsFromGLTF(device, processed);
```

I'm not sure exactly what to do to the scenegraph to replace the texture with a different image. I see the there is a texCoord buffer in the ModelNode's model:
![image](https://github.com/user-attachments/assets/f712ef7b-a5d1-4ea3-bad2-492d07f01238)
But I'm not sure what buffer or accessor the image is stored in?

There is some related documentation on loaders.gl that shows recreating the gltf file [here](https://loaders.gl/docs/modules/gltf/api-reference/gltf-scenegraph#modifying), but I cannot seem to get the vertex buffer to be the right size when recreating, so I was hoping to do it after.

And if I can replace it, how do I then make the scenegraph update to reflect this?

Thankyou for any help!

**Comment by ibgreen** [comment]
@matthew-rimmer As a quick answer: Have you looked at the `GLTFLoader` docs? https://loaders.gl/docs/modules/gltf/api-reference/gltf-loader#data-format.

It shows you that the loaded images are decoded and stored in the `<result>.images` array. If you can identify the right index into that array, you should simply be able to switch out the image before calling `postProcessGLTF()`.
